plotly/plotly/plotly.py

```diff
@@ -1449,6 +1454,194 @@ def _send_to_plotly(figure, **plot_options):
+    if url_path[-1] == '/':
+        url_path = url_path[0: -1]
+    url_path = url_path.replace('/', ':')
+
+    meta_get_url = upload_url + '/' + url_path
+    get_url = meta_get_url + '/content'
+
+    r = requests.get(get_url, headers=headers)
+    json_res = json.loads(r.text)
+
+    # make request to grab the grid id (fid)
+    r_meta = requests.get(meta_get_url, headers=headers)
+    json_res_meta = json.loads(r_meta.text)
+    retrieved_grid_id = json_res_meta['fid']
+
+    if raw is False:
+        return Grid(json_res, retrieved_grid_id)
+    else:
+        return json_res
+
+
+def create_animations(figure, filename=None, sharing='public', auto_open=True):
+    """
+    BETA function that creates plots with animations via `frames`.
+
+    Creates an animated plot using 'frames' alongside 'data' and 'layout'.
+    This BETA endpoint is subject to deprecation in the future. In relation
+    to `plotly.plotly.plot`, folder-creation and overwriting are not supported
+    but creating a plot with or without animations via frames is supported.
+
+    :param (str) filename: if set to 'None', an automatically generated plot
+        name will be created. Does not support folder creation, meaning that
+        a folder of the form 'folder/name' will NOT create a the folder and
+        place the plot in it.
+    :param (str) sharing: see `plotly.plotly.plot()` doc string.
+    :param (bool) auto_open: if True, opens plot in the browser. If False,
+        returns the url for the plot instead.
+
+    Example 1: Simple Animation
+    ```
+    import plotly.plotly as py
+    from plotly.grid_objs import Grid, Column
+
+    column_1 = Column([1, 2, 3], 'x')
+    column_2 = Column([1, 3, 6], 'y')
+    column_3 = Column([2, 4, 6], 'new x')
+    column_4 = Column([1, 1, 5], 'new y')
+    grid = Grid([column_1, column_2, column_3, column_4])
+    py.grid_ops.upload(grid, 'animations_grid', auto_open=False)
+
+    # create figure
+    figure = {
+        'data': [
+            {
+                'xsrc': grid.get_fid_uid('x'),
+                'ysrc': grid.get_fid_uid('y')
+            }
+        ],
+        'layout': {'title': 'First Title'},
+        'frames': [
+            {
+                'data': [
+                    {
+                        'xsrc': grid.get_fid_uid('new x'),
+                        'ysrc': grid.get_fid_uid('new y')
+                    }
+                ],
+                'layout': {'title': 'Second Title'}
+            }
+        ]
+    }
+
+    py.create_animations(figure, 'new_plot_with_animations')
+    ```
+    """
+    credentials = get_credentials()
+    validate_credentials(credentials)
+    username, api_key = credentials['username'], credentials['api_key']
+    auth = HTTPBasicAuth(str(username), str(api_key))
+    headers = {'Plotly-Client-Platform': 'python',
+               'content-type': 'application/json'}
+
+    json = {
+        'figure': figure,
+        'world_readable': True
```

Comment:
I don't think it makes sense to make set `sharing` to `private` as a default if the Community users get only 1. I think that may be confusing and more work if they have to set it to `public` each time they make a plot.